gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -271,4 +272,15 @@ private static String encodeCertificate(Certificate palantirCert) {
             throw new RuntimeException("Could not convert Palantir cert back to regular", e);
         }
     }
+
+    public static String getSerialNumber(String certContent) {
+        try {
+            InputStream in = new ByteArrayInputStream(certContent.getBytes(StandardCharsets.UTF_8));
+            CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+            X509Certificate cert = (X509Certificate) certFactory.generateCertificate(in);
+            return cert.getSerialNumber().toString();
+        } catch (CertificateException e) {
+            throw new RuntimeException(e);
```

Comment:
Best to add "could not get serial number for certificate" + `certContent` so we can more easily debug.